pkg/util/k8sutil/k8sutil.go

```diff
@@ -45,6 +46,21 @@ func init() {
 	cgoscheme.AddToScheme(scheme)
 }
 
+// UtilDecoderFunc finds the correct decoder from a GroupVersion
+type UtilDecoderFunc func(gv schema.GroupVersion) runtime.Decoder
+
+// SetDecoderFunc sets a non default decoder function
+// This is used as a work around to add support for unstructured objects
+func SetDecoderFunc(u UtilDecoderFunc) {
+	decoderFunc = u
+}
+
+// GetCodecs get the codec factory for the scheme
+// Exposing the codecs factory so that a user can use the codecs for the scheme.
+func GetCodecs() serializer.CodecFactory {
```

Comment:
If this is unacceptable, then we could add the codecs as a parameter to the `UtilDecoderFunc`. @hasbro17 I would love to hear your thoughts on which approach?

Reply:
@shawn-hurley Let's pass in the CodecFactory as an argument to `UtilDecoderFunc`. That way `SetDecoderFunc` is less ambiguous to use and we don't increase the API surface if we really don't need to.

So we can get rid of `GetCodecs()` and change `UtilDecoderFunc` to
```Go
type UtilDecoderFunc func(codecFactory serializer.CodecFactory, gv schema.GroupVersion) runtime.Decoder
```

You'll have to change the defaultDecoder function signature and wherever it's called below as well
```Go
func decoder(codecFactory serializer.CodecFactory, gv schema.GroupVersion) runtime.Decoder {
	return codecFactory.UniversalDecoder(gv)
 }
```

```Go
func RuntimeObjectFromUnstructured(u *unstructured.Unstructured) runtime.Object {
	gvk := u.GroupVersionKind()
	decoder := decoderFunc(codecs, gvk.GroupVersion())
```